lib/light_accessory.js

```diff
@@ -75,13 +75,12 @@ class LightAccessory extends BaseAccessory {
         this.normalAsync(Characteristic.Hue, hue, this.isRefesh)
 
         var saturation;
-        saturation = Math.floor((this.colourObj.s - this.function_dp_range.saturation_range.min) * 100 / (this.function_dp_range.saturation_range.max - this.function_dp_range.saturation_range.min));  // saturation 0-100
+        saturation = Math.floor((this.colourObj.s - this.function_dp_range.saturation_range.min) * 100 / (this.function_dp_range.saturation_range.max - this.function_dp_range.saturation$
```

Comment:
Please restore the previous code `saturation = Math.floor((this.colourObj.s - this.function_dp_range.saturation_range.min) * 100 / (this.function_dp_range.saturation_range.max - this.function_dp_range.saturation_range.min));  // saturation 0-100`.

Because there is an obvious bug. `this.function_dp_range.saturation$`

Reply:
Got it. I don't think I removed this, but added back in.